raiden/transfer/events.py

```diff
@@ -386,27 +386,30 @@ def __init__(
             identifier: PaymentID,
             amount: TokenAmount,
             target: TargetAddress,
+            secret: Secret = None,
```

Comment:
Note: If `None` is made into an invalid value here a migration is necessary.

Reply:
The reason for the Secret = `None` is a few tests which are using this object without providing the secret (since this is new). I can set the default value to be EMPTY_SECRET and remove the `if` from to_dict(). Alternatively can add a random_secret to the tests.

If you like to avoid the `if` from the from_dict() too, migration is necessary. If this is the prefered option, I suggest to implement as a separate task (I can do it if needed).

Reply:
This can be fixed in a follow up PR